BAZProject/Box.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  Box.swift
+//  BAZProject
+//
+//  Created by Manuel S on 18/04/23.
+//
+
+import Foundation
+
+//MARK: Box para manejo de MVVM
+public struct Box<T> {
+    public var value: T {
+        didSet {
+            listener?()
+        }
+    }
+    var listener: (() -> Void)?
+    
+    public init(value: T) {
+        self.value = value
+    }
+    
+    public mutating func bind(_ listener: @escaping (() -> Void)) {
```

Comment:
Tal vez podrías usar un typealias para mejorar la lectura de tus métodos. Por ejemplo:

`typealias CompletionHandler = () -> Void`

Y así poder usarlo así:
```suggestion
    public mutating func bind(_ listener: @escaping CompletionHandler) {
```